scripts/github-release.js

```
@@ -0,0 +1,102 @@
+  if (!tagsResponse.ok) {
+    throw new Error(`Failed to fetch tags. ${tagsResponse.statusText}`);
+  }
+  const tags = await tagsResponse.json();
+
+  const commitsResponse = await fetch(`${root}/commits`, {headers});
+  if (!commitsResponse.ok) {
+    throw new Error(`Failed to fetch commits. ${commitsResponse.statusText}`);
+  }
+  const [releaseCommit, ...commits] = await commitsResponse.json();
+
+  const releaseMessageRgx = /^Release (v[0-9]+\.[0-9]+\.[0-9]+)/;
+  const match = releaseMessageRgx.exec(releaseCommit.commit.message);
+  if (!Array.isArray(match) || typeof match[1] !== 'string') {
+    throw new Error(
+      `Failed to parse release commit message: ${releaseCommit.commit.message}`,
+    );
+  }
+  const releaseVersion = match[1];
+
+  const commitsSinceTag = [];
+  for (let commit of commits) {
+    if (commit.sha === tags[0].commit.sha) {
+      break;
+    }
+    commitsSinceTag.push(commit);
+  }
+  const prs = await Promise.all(commitsSinceTag.map(c => prFromCommit(c.sha)));
+
+  const changelogEntries = [];
+  for (let i = 0; i < commitsSinceTag.length; i++) {
+    const commit = commitsSinceTag[i];
+    const pr = prs[i];
+    if (pr) {
+      changelogEntries.push(`- ${commit.commit.message} (#${pr})`);
+    } else {
+      changelogEntries.push(`- ${commit.commit.message}`);
+    }
+  }
+
+  const options = {
+    method: 'POST',
+    headers: {
+      ...headers,
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      tag_name: releaseVersion,
+      name: releaseVersion,
+      body: '## Changelog\n' + changelogEntries.join('\n'),
+    }),
+  };
+
+  const releaseResponse = await fetch(`${root}/releases`, options);
+  if (!releaseResponse.ok) {
+    throw new Error(`Failed to create release. ${releaseResponse.statusText}`);
+  }
+
+  const release = await releaseResponse.json();
+  console.log(`Successfully created release at: ${release.html_url}`);
+}
+
+main();
```

```suggestion
main().catch(console.error);
```

not sure if our node.js version would hide the errors 🤔